src/coreclr/src/utilcode/clrhost.cpp

```diff
@@ -12,7 +12,20 @@
 #include "clrnt.h"
 #include "contract.h"
 
-HINSTANCE g_hmodCoreCLR;
+#if HOST_WINDOWS
+extern "C" IMAGE_DOS_HEADER __ImageBase;
+#endif
+
+void* GetClrModuleBase()
+{
+    LIMITED_METHOD_CONTRACT;
+
+#if HOST_WINDOWS
+    return (void*)&__ImageBase;
+#else // !HOST_UNIX
+    return (void*)PAL_GetSymbolModuleBase((void*)GetClrModuleBase);
```

Comment:
`PAL_GetSymbolModuleBase` likely involves taking locks and walking list of all modules in the process. Do we need to cache the result?

Reply:
Yes. It looks like a typical implementation (i.e. glibc) is a hash lookup while holding a lock to prevent load/unload. 
We can trivially cache this though, so it is a good idea. 👍